### DISCUSSION TITLE
[SOLVED] Confirmation for "Clean all" shortcut

### DISCUSSION THREAD

**Discussion by slalaurette** [martinrotter/rssguard#1005] [Ideas]
I was testing my keyboard shortcuts and, to my surprise, the shortcut for "Clean all" (which deletes all articles in all feeds) works instantly, without confirmation. So I suddenly got a completely empty list. 

It's OK, I was testing so I was willing to lose all articles, but I think this is dangerous for a user who is just using the application and might press the key combo by mistake. Maybe RSS Guard should ask for confirmation before cleaning the whole board?

**Comment by martinrotter** [comment]
If you hit clean all, then those articles are moved to recycle bin and can be easily restored by right clicking recycle bin.

If you use "clean selected" action and recycle bin is selected, then it is emptied and articles disappear from the bin.

I agree that I could add confirmation box when emptying recycle bin. I will add it soon.

**Comment by slalaurette** [comment]
Thank you! I still think the confirmation step would need to happen before deleting, because if your recycle bin is already crowded, and you clean all items by mistake, then restoring the recycle bin would also restore all the stuff you had deleted on purpose.

**Reply by martinrotter** [reply]
Yes, I am adding confirmations for all cleanups right now.
